test/integration/targets/mysql_db/tasks/state_dump_import.yml

```
@@ -58,12 +73,68 @@
 - name: state dump/import - file name should exist
   file: name={{ db_file_name }} state=file
 
+- name: state dump with multiple databases in comma separated form.
+  mysql_db:
+    name: "{{ db_name }},{{ db_name2 }}"
+    state: dump
+    target: "{{ dump_file1 }}"
+    login_unix_socket: '{{ mysql_socket }}'
+  register: dump_result1
+
+- name: assert successful completion of dump operation (with multiple databases in comma separated form)
+  assert:
+    that:
+       - "dump_result1.changed == true"
+
+- name: state dump - dump file1 should exist
+  file: name={{ dump_file1 }} state=file
+
+- name: state dump with multiple databases in list form.
+  mysql_db:
+    name:
+      - "{{ db_name }}"
+      - "{{ db_name2 }}"
+    state: dump
+    target: "{{ dump_file2 }}"
+    login_unix_socket: '{{ mysql_socket }}'
+  register: dump_result2
+
+- name: assert successful completion of dump operation (with multiple databases in list form)
+  assert:
+    that:
+       - "dump_result2.changed == true"
+
+- name: state dump with multiple databases in list form via check_mode
+  mysql_db:
+    name:
+      - "{{ db_name }}"
+      - "{{ db_name2 }}"
+    state: dump
+    target: "{{ dump_file2 }}"
+    login_unix_socket: '{{ mysql_socket }}'
+  register: dump_result
+  check_mode: yes
+
+- name: assert successful completion of dump operation (with multiple databases in list form) via check mode
+  assert:
+    that:
+       - "dump_result.changed == true"
+
+- name: state dump - dump file2 should exist
```

```suggestion
- name: state dump - dump file2 should exist
```
Probably we should remove files previously, then check that it does not exist after check_mode execution

---

OR call check_mode before the actual dump operation and then check that dump_file does not exist.

Good catch